patches/Avalon/Terraria/Main.cs.patch

```diff
@@ -1,23 +1,187 @@
++		private static Color[] _mapColorCacheArray = new Color[30000];
+ 		public WaterfallManager waterfallManager;
+ 		public static WorldSections sectionManager;
+ 		public static bool ServerSideCharacter;
 @@ -293,12 +_,12 @@
  		public static int dedServCount2;
  		public static bool superFast = false;
  		public const int offLimitBorderTiles = 40;
 -		public const int maxItemTypes = 5045;
-+		public const int maxItemTypes = ItemID.Count;
- 		public const int maxProjectileTypes = 950;
- 		public const int maxNPCTypes = 663;
- 		public const int maxTileSets = 623;
- 		public const int maxWallTypes = 316;
++		public const int maxItemTypes = ItemID.Count; // TODO: This is only the beginning of the stupid fucking maxXType variables
+-		public const int maxProjectileTypes = 950;
++		public const int maxProjectileTypes = 1140;
+-		public const int maxNPCTypes = 663;
++		public const int maxNPCTypes = 775;
+-		public const int maxTileSets = 623;
++		public const int maxTileSets = 775;
+-		public const int maxWallTypes = 316;
++		public const int maxWallTypes = 468;
 -		public const int maxBuffTypes = 323;
 +		public const int maxBuffTypes = BuffID.Count;
  		public const int maxGlowMasks = 301;
  		public const int maxExtras = 212;
  		public const int maxGoreTypes = 1269;
-@@ -518,15 +_,15 @@
+@@ -319,7 +_,7 @@
+ 		public const int numAccBalloon = 18;
+ 		public const int maxWings = 47;
+ 		public const int maxBackgrounds = 298;
+-		public const int MaxShopIDs = 100;
++		public const int MaxShopIDs = 100; // TODO: This isn't the end :weary:
+ 		private static float cameraLerp;
+ 		private static int cameraLerpTimer;
+ 		private static int cameraLerpTimeToggle;
+@@ -410,7 +_,7 @@
+ 		public static bool stopTimeOuts = false;
+ 		public static bool showSpam = false;
+ 		public static bool showItemOwner = false;
+-		public static bool[] townNPCCanSpawn = new bool[663];
++		public static bool[] townNPCCanSpawn = new bool[663]; // TODO: This is an array that needs to be changed
+ 		public static int musicBox2 = -1;
+ 		public static byte HealthBarDrawSettings = 1;
+ 		public static bool runningCollectorsEdition;
+@@ -457,6 +_,8 @@
+ 		public static int[] snowBG = new int[3];
+ 		public static int[] hallowBG = new int[3];
+ 		public static int[] crimsonBG = new int[3];
++		public static int contagionBG = 185;
++		public static int darkBG = 186;
+ 		public static int[] desertBG = new int[3];
+ 		public static int[] mushroomBG = new int[3];
+ 		public static int oceanBG;
+@@ -516,17 +_,17 @@
+ 		public static int buffScanAreaWidth = (maxScreenW + 800) / 16 - 1;
+ 		public static int buffScanAreaHeight = (maxScreenH + 800) / 16 - 1;
  		public static float musicPitch = 0f;
- 		public static bool[] projHostile = new bool[950];
+-		public static bool[] projHostile = new bool[950];
++		public static bool[] projHostile = new bool[950]; // TODO: haha more arrays that need to be changed
```

Comment:
Why didn't you change it yet then? You've already down the IDs, haven't you?

Reply:
because I didn't think we should change it until the actual content was added? I didn't know if it would break or not